### PR TITLE
Update position and orientation when motion state is updated

### LINKED ISSUE #217: Route user entity state updates from systems to User::entity_state
Currently when a user's entity is updated, such as position, that state update is not visible to the outside world.

To fix this there needs to be a window into the current state, or that state needs to be updated when looking through the current window.

### PR DESCRIPTION
**Pull request by adam4813**
Read state data from Multiton instead of game state in RenderSystem.

Properly handle position and orientation loading/unloading in PhysicsSystem.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

Closes #217

### CODE REVIEW

File: common/components/collision-body.hpp [hunk 1]
```diff
@@ -5,32 +5,60 @@
...
+				if (OrientationMap::Has(entity_id)) {
+					auto orientation = OrientationMap::Get(entity_id);
+					btQuaternion rotation = transform.getRotation();
+					orientation->value.x = rotation.x();
+					orientation->value.y = rotation.y();
+					orientation->value.z = rotation.z();
+					orientation->value.w = rotation.w();
+					rotation.getEulerZYX(orientation->rotation.z, orientation->rotation.y, orientation->rotation.x);
+				}
+			}
```
Comment (Meisaka) [inline common/components/collision-body.hpp:53]: At the end of the physics system's `Update` method is a loop that maps entity IDs to transforms, if you moved all this there, then the motion state wouldn't have to track an entity ID, and all the positions and orientations would still get updated at the end of a step.

**Review by Meisaka — APPROVED** [review]
Just one minor concern, otherwise looks good.

File: common/components/collision-body.hpp (hunk 1 above)
Comment (adam4813) [reply common/components/collision-body.hpp:53]: I did have it there, but it felt odd.

The btMotionState docs recommended syncing the state in the method. The transform_updated bool was a hack to do this sync to the GameState.

File: server/main.cpp [hunk 2]
```diff
@@ -83,7 +83,7 @@ int main() {
 
 		tec::SaveGame save;
 		save.Load(tec::FilePath::GetAssetPath("save/save1.json"));
-		lua_sys->GetGlobalState()["save"] = save;
+		lua_sys->GetGlobalState()["save"] = &save;
```
[the remark this answers is not in the document]
Comment (adam4813) [reply server/main.cpp:86]: For some reason when it copied the save the UserList would copy ok, but the entity_id of each User in the list did not.

There is an underlying issues there, but this was also the more correct course of action.

**Pull request merged**